product-accelerators/apim/components/org.wso2.healthcare.apim.backendauth/src/main/java/org/wso2/healthcare/apim/backendauth/BackendAuthenticator.java

```diff
@@ -56,28 +62,35 @@ public boolean mediate(MessageContext messageContext) {
             log.debug("Backend authenticator mediator is invoked.");
         }
 
-        BackendAuthConfig config = backendAuthConfig.get(configName);
-        String accessToken;
-        BackendAuthHandler currentAuthenticator;
+        if (masterConfig.getClientId() != null && masterConfig.getClientId().contains("$")) {
+            masterConfig.setClientId(Utils.resolveConfigValues(this.clientId, messageContext));
+        }
+        if (masterConfig.getPrivateKeyAlias() != null && masterConfig.getPrivateKeyAlias().contains("$")) {
+            masterConfig.setPrivateKeyAlias(Utils.resolveConfigValues(this.keyAlias, messageContext));
+        }
 
-        if (configName == null) {
-            log.error("Auth type is not defined in the message context.");
+        if (!Utils.validateConfig(masterConfig)) {
```

Comment:
Can't we validate the config whenever the setter is called?

Reply:
there are multiple setters (for each inline config) and the allowed combinations can only be validated at the request flow. (setters are calling sequentially and validation need to happen in the last call; we don't have a way to know the last call)

Reply:
+1